analyses/survival-analysis/survival-analysis_histologies.Rmd

```diff
@@ -0,0 +1,234 @@
+```
+
+Run on all PBTA and HGAT only
+## Kaplan-Meier
+
+The [Kaplan-Meier](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3059453/) 
+survival curve shows the probability of survival over time.
+
+```{r Kaplan Meier}
+
+# let's look at HGAT
+meta_all <- meta_indep
+meta_hgat <- meta_indep %>%
+  filter(short_histology == "HGAT")
+meta_no_hgat <- meta_indep_no_hgat
+
+as.data.frame(table(meta_indep$short_histology))
+
+prefix_list <- c("all-pbta","hgat","no-hgat")
+# define data frame list in the same order as prefix list 
+df_list <-list(meta_all, meta_hgat, meta_no_hgat)
+  
+for (each in phenotypes) {
+  for(i in 1:length(prefix_list)){
+    
+    prefix <- prefix_list[i]
+    df_of_interest <- df_list[[i]]
+    
+    kap_fit <- survival_analysis(df_of_interest,
+                             ind_var = each,
+                             test = "kap.meier",
+                             metadata_sample_col = "Kids_First_Biospecimen_ID_RNA"
+    )
+    
+    # median and 0.95CI survival days
+    kap_fit$model
+    
+    #`survminer` package comes with a convenient plotting function which uses `ggplot2` arguments. 
+    #It requires the original Kaplain-Meier `survfit` object, so we will extract that with `$model`.
+    surv_plot <- survminer::ggsurvplot(kap_fit$model,
+                                       pval = TRUE,
+                                       data = kap_fit$original_data,
+                                       risk.table = TRUE,
+                                       xlim = c(0, 8000),
+                                       break.time.by = 1000,
+                                       ggtheme = theme_pubr(),
+                                       risk.table.y.text.col = TRUE,
+                                       risk.table.y.text = FALSE,
+                                       palette = each
+                                       #color = phenotype,
+                                       #palette = "Paired"
+    
+    )
+    
+    surv_plot$plot <- surv_plot$plot +
+      #ggtitle(paste0(ptitle)) +
+      theme(legend.position = "right")
+    
+    # Make this plot a meta_indep plot
+    p <- cowplot::plot_grid(surv_plot[[1]], surv_plot[[2]], nrow = 2,
+                         rel_heights = c(4, 2))
+    
+    # Print it out here
+    print(p)
+    
+    #Save the plot to a file. 
+    ggsave(filename = file.path(plots_dir, paste(prefix, each, "KM.pdf", sep = "-")), plot = p)
+    # Let's save the model for CI and median survival as well
+    readr::write_tsv(surv_median(kap_fit$model), file.path(results_dir, paste(prefix, each, "model.tsv", sep = "-")))
+    }
+  }
+
+```
+
+#### output statistics for paper writing 
+
+```{r}
+# first format the metadata to have appropriate OS_status column
+df_list <- lapply(df_list, format_meta)
+
+# define the formula
+model <- as.formula("survival::Surv(OS_days, OS_status) ~ pheno_0.5")
```

Comment:
yeah good point - @runjin326 can you adjust this to print all model stats? I think I had this one in the notebook because I was thinking we would use that as the main figure (to be discussed), but we should add all so we can also describe supplemental figures